packages/framework/client-logger/external-telemetry/src/container/containerTelemetry.ts

```diff
@@ -0,0 +1,170 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { IExternalTelemetry } from "../common/index.js";
+import { ICriticalContainerError } from "@fluidframework/container-definitions";
+
+/**
+ * This file contains the types for container telemetry that can be produced.
+ */
+
+/**
+ * This object contains names for Container Telemetry. Unlike the raw container system event names they contain more information such as the scope
+ *
+ * @beta
+ */
+export const ContainerTelemetryEventNames = {
+	/**
+	 * Name for the container telemetry event that is intended to be produced from the internal container "connected" system event
+	 *
+	 * @see {@link ContainerConnectedTelemetry}
+	 */
+	CONNECTED: "container.connected",
+	/**
+	 * Name for the container telemetry event that is intended to be produced from the internal container "disconnected" system event
+	 *
+	 * @see {@link ContainerDisconnectedTelemetry}
+	 */
+	DISCONNECTED: "container.disconnected",
+	/**
+	 * Name for the container telemetry event that is intended to be produced from the internal container "closed" system event
+	 *
+	 * @see {@link ContainerClosedTelemetry}
+	 */
+	CLOSED: "container.closed",
+	/**
+	 * Name for the container telemetry event that is intended to be produced from the internal container "attaching" system event
+	 *
+	 * @see {@link ContainerAttachingTelemetry}
+	 */
+	ATTACHING: "container.attaching",
+	/**
+	 * Name for the container telemetry event that is intended to be produced from the internal container "attached" system event
+	 *
+	 * @see {@link ContainerAttachedTelemetry}
+	 */
+	ATTACHED: "container.attached",
+} as const;
+
+/**
+ * The type for all values within {@link ContainerTelemetryEventNames}
+ * @beta
+ */
+export type ContainerTelemetryEventName =
+	(typeof ContainerTelemetryEventNames)[keyof typeof ContainerTelemetryEventNames];
+
+/**
+ * The base interface for all Container telemetry
+ * @beta
+ */
+export interface IContainerTelemetry extends IExternalTelemetry {
+	/**
+	 * @inheritdoc IExternalTelemetry.eventName
+	 */
+	eventName: ContainerTelemetryEventName;
+	/**
+	 * The client side generated unique id of the container this telemetry came from.
+	 */
+	containerId: string;
+	/**
+	 * { @inheritdoc @fluidframework/container-definitions#IContainer."clientId" }
+	 */
+	clientId?: string;
+	/**
+	 * The id of the container this resolved url is for.
+	 * Will be undefined only when the container is not attached.
```

Comment:
```suggestion
	 * Unique identifier for a container, stable across creation and load.
	 * I.e. different clients loading the same container (or the same client loading the container two separate times)
	 * will agree on this value.
	 *
	 * remarks This can be undefined for a container that has not been attached.
	 *
	 * @see {@link IContainerTelemetry.containerId | `containerId` for an identifier of a particular _instance_ of the container being created/loaded }
```